### PR TITLE
updated backend to read 3box collectiblesFavorite

### PR DESCRIPTION
**Pull request by dan13ram**

### CODE REVIEW

File: packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts [hunk 1]
```diff
@@ -47,3 +50,13 @@ function getImage(
   }
   return '';
 }
+
+function getCollectiblesFavourites(
+  collectiblesFavorites: Array<CollectiblesFavorite> | null | undefined,
```
Comment (pakokrew) [inline packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts:55]: the schema is
`collectiblesFavorites: [CollectiblesFavorites!]`

so maybe the return type should be only Array<CollectiblesFavorite> and always return an empty array if theres nothing in the profile

File: packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts [hunk 2]
```diff
@@ -34,6 +34,9 @@ export const getBoxProfile: QueryResolvers['getBoxProfile'] = async (
       height: 300,
     }),
     website: boxProfile.website,
+    collectiblesFavorites: getCollectiblesFavourites(
+      boxProfile?.collectiblesFavorites,
```
Comment (pakokrew) [inline packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts:38]: maybe that `?` is not necessary here

**Review by pakokrew — CHANGES_REQUESTED** [review]

File: packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts [hunk 3]
```diff
@@ -47,3 +50,13 @@ function getImage(
   }
   return '';
 }
+
+function getCollectiblesFavourites(
+  collectiblesFavorites: Array<CollectiblesFavorite> | null | undefined,
+) {
+  if (!collectiblesFavorites) return null;
```
Comment (dan13ram) [inline packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts:57]: @pakokrew its handled here. The schema allows for null value to be returned.

File: packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts (hunk 1 above)
Comment (dan13ram) [reply packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts:55]: so do you want me to change the schema to `collectiblesFavorites: [CollectiblesFavorites!]!`

Comment (dan13ram) [reply packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts:55]: I wanted the schema to remain uniform with the rest of the values:
```
  type BoxProfile {
    ethereumAddress: String
    name: String
    description: String
    location: String
    job: String
    emoji: String
    imageUrl: String
    coverImageUrl: String
    website: String
    collectiblesFavorites: [CollectiblesFavorites!]
  }
```

Comment (dan13ram) [reply packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts:55]: also note the `type CollectiblesFavorite` in this line is different from our typeDef, and comes from 3box.

Comment (pakokrew) [reply packages/backend/src/handlers/remote-schemas/resolvers/getBoxProfile/resolver.ts:55]: Ok my bad, didn't see that all the properties were nullable in the GraphQL schema. It's ok like this then !

**Review by pakokrew — APPROVED** [review]
LGTM

**Pull request merged**